frontend/src/components/Table/VirkarekisteriTable.tsx

```diff
@@ -93,70 +110,140 @@ const DataTable: React.FC = () => {
+          ? (position.vacancyNumber?.includes(vacancyNumberSearch) ?? false)
+          : true;
+        const matchesSijoituspaikka = placementLocationStateSearch
+          ? (position?.placementLocation?.includes(placementLocationStateSearch) ?? false)
+          : true;
+        return matchesVakanssinumero && matchesSijoituspaikka;
+      }
+    });
+    setFilteredData(filtered);
+  };
+
+  const handleReset = () => {
+    setVacancyNumberSearch('');
+    setPlacementLocationSearch('');
+    setFilteredData(dataFromBackend);
+  };
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0); // Reset to first page on rows per page change
+  };
+
   const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = useTable<Position>(
-    { columns, data: dataFromBackend },
+    { columns, data: filteredData },
     useSortBy,
   );
-
+  const paginatedRows = rows.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage);
   return (
-    <TableContainer component={Box} sx={{ margin: 2, border: '1px solid #ccc' }}>
-      <Table {...getTableProps()} sx={{ minWidth: 650 }}>
-        <TableHead sx={{ backgroundColor: '#223B7C', height: '30px' }}>
-          {headerGroups.map((headerGroup) => (
-            <TableRow {...headerGroup.getHeaderGroupProps()}>
-              {/* eslint-disable-next-line @typescript-eslint/no-explicit-any*/}
-              {headerGroup.headers.map((column: any) => (
-                <TableCell
-                  {...column.getHeaderProps(column.getSortByToggleProps())}
+    <>
+      <Accordion>
+        <AccordionSummary expandIcon={<ExpandMoreIcon />} sx={{ backgroundColor: alpha('#223B7C', 1), color: 'white' }}>
+          <Typography sx={{ color: 'white' }}>{t('search_filter.search_filters')}</Typography>
+        </AccordionSummary>
+        <AccordionDetails>
+          <Box display="flex" flexDirection="column" gap={2}>
+            <TextField
+              label={t('table.vacancy_number')}
+              value={vacancyNumberSearch}
+              onChange={(e) => setVacancyNumberSearch(e.target.value)}
+              fullWidth
+            />
+            <TextField
+              label={t('table.placement_location')}
+              value={placementLocationStateSearch}
+              onChange={(e) => setPlacementLocationSearch(e.target.value)}
+              fullWidth
+            />
+            <Button variant="contained" onClick={handleSearch}>
+              {t('search_filter.search')}
+            </Button>
+            <Button variant="outlined" onClick={handleReset}>
+              {t('search_filter.reset')}
+            </Button>
+          </Box>
+        </AccordionDetails>
+      </Accordion>
+      <TableContainer component={Box} sx={{ margin: 2, border: '1px solid #ccc' }}>
```

Comment:
```suggestion
      <TableContainer component={Box} sx={{ border: '1px solid #ccc' }}>
```
![image](https://github.com/user-attachments/assets/619ae79e-6d9c-4f92-ac0b-009589e1691e)

I think we should remove the margin from the table in this ticket so all components are aligned nicely

Reply:
Will be removed with other PR